core-plugins/src/main/java/io/cdap/plugin/batch/source/XMLReaderBatchSource.java

```diff
@@ -137,6 +137,10 @@ public void prepareRun(BatchSourceContext context) throws Exception {
       conf.set(XMLInputFormat.XML_INPUTFORMAT_TARGET_FOLDER, config.targetFolder);
     }
 
+    if (!config.containsMacro("enableExternalEntities") && !Strings.isNullOrEmpty(config.enableExternalEntities)) {
```

Comment:
No need to check for macro here, it will always be evaluated at this point.

Should also remove the null or empty check and have a getter method in the config class that checks for null like:

```
public boolean shouldEnableExternalEntities() { 
  return enableExternalEntities == null ? false : true
}
```

This removes the possibility of unintended null pointer exceptions. Then you can simplify this to one line:

```
conf.set(..., String.valueOf(config.shouldEnableExternalEntities()));
```